lib/ex_doc/formatter/markdown/templates/module_template.eex

```diff
@@ -0,0 +1,35 @@
+# <%= module.title %> <%= module_type(module) %> (<%= config.project %> v<%= config.version %>)
+
+<%= for annotation <- module.annotations do %>*(<%= annotation %>)* <% end %>
+
+<%= if deprecated = module.deprecated do %>
+> This <%= module.type %> is deprecated. <%=h deprecated %>.
+
+<% end %>
+<%= if doc = node_doc(module) do %>
+<%= doc %>
+<% end %>
+
+<%= if summary != [] do %>
+## Table of Contents
+<%= for {name, nodes} <- summary, do: summary_template(name, nodes) %>
+<% end %>
+
+## Contents
+
+<%= for {name, nodes} <- summary, _key = text_to_id(name) do %>
+### <%=h to_string(name) %>
+
+<%= for node <- nodes do %>
+<%= detail_template(node, module) %>
+<% end %>
+
+<% end %>
```

Comment:
I would start with a much simpler version for Markdown, something like this:

```suggestion
<%= for {_name, nodes} <- summary, node <- nodes do %>
  <%= detail_template(node, module) %>
<% end %>
```

My thinking is that, because we tell users they can use `##` in their docs, our only option is to use `#` for the functions, types, macros, etc.